### PR TITLE
feat: added support for php package

### LINKED ISSUE #1007: Add php (simple) release-type
I found a problem when trying to run Actions in my php project. This is a project that contains one composer.json file. The current php-youshi configuration does not work correctly with such projects: for example, the description of commits is not added to CHANGELOG.md, and the version in composer.json in the project root is not changed. I researched the problem and found that the php-yoishi configuration expects a lot of specific project files, and is a little bit incompatible with regular php projects.

I want to suggest adding one more release-type - php. I have prepared a code that can be viewed at the pull request 

### my new release-type php
```
~/my/release-please(php_package_support) » ./build/src/bin/release-please.js release-pr --token=$GITHUB --repo-url="setnemo/php-star-dot-notation" --release-type="php"
✔ found 1 commits since b4726d9ceda620e9b5bf03c23b20a6415b5097e4
✔ release as patch: There are 0 BREAKING CHANGES and 0 features
✔ updating version from 0.0.1 to 2.0.1
✔ updating CHANGELOG.md
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "376ede538a3bf4a00447d5965800cb6a10efbaa2".
✔ Skipping branch creation step...
✔ Got the latest commit tree
✔ Successfully created a tree with the desired changes with SHA c4d36481eeef420b56b0576a84a2d72f718a187e
✔ Successfully created commit. See commit at https://api.github.com/repos/setnemo/php-star-dot-notation/git/commits/edf84621f3dd04761a87e9bc7d8c713471835a44
✔ Successfully updated reference release-v2.0.1 to edf84621f3dd04761a87e9bc7d8c713471835a44
✔ Successfully opened pull request available at url: https://api.github.com/repos/setnemo/php-star-dot-notation/pulls/16.
✔ Successfully opened pull request: 16.
✔ adding label autorelease: pending to https://github.com/setnemo/php-star-dot-notation/pull/16
✔ find stale PRs with label "autorelease: pending"
```
PR was created 
PR with information from commits
PR with information from commits

### Existing release-type php-yoshi 
```
~/my/release-please(php_package_support) » ./build/src/bin/release-please.js release-pr --token=$GITHUB --repo-url="setnemo/php-star-dot-notation" --release-type="php-yoshi"
✔ found 1 commits since b4726d9ceda620e9b5bf03c23b20a6415b5097e4
✔ release as patch: There are 0 BREAKING CHANGES and 0 features
✔ no updates necessary for composer.json
⚠ file docs/manifest.json did not exist
✔ updating CHANGELOG.md
⚠ file src/Version.php did not exist
⚠ file src/ServiceBuilder.php did not exist
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "376ede538a3bf4a00447d5965800cb6a10efbaa2".
✔ Skipping branch creation step...
✔ Got the latest commit tree
✔ Successfully created a tree with the desired changes with SHA 3ec10df06c6954e8f0d1582b21cc54a4e891b749
✔ Successfully created commit. See commit at https://api.github.com/repos/setnemo/php-star-dot-notation/git/commits/4cdffa407f8adc0ce50064a6ae0fd3b0d6875e9f
✔ Successfully updated reference release-v2.0.1 to 4cdffa407f8adc0ce50064a6ae0fd3b0d6875e9f
✔ Successfully opened pull request available at url: https://api.github.com/repos/setnemo/php-star-dot-notation/pulls/17.
✔ Successfully opened pull request: 17.
✔ adding label autorelease: pending to https://github.com/setnemo/php-star-dot-notation/pull/17
✔ find stale PRs with label "autorelease: pending"
```
PR was created 
PR with information from commits
PR with information from commits

### PR DESCRIPTION
**Pull request by setnemo**
php-yoshi doesn't work correctly with php projects without specific files from existing configuration

Also refactored existing configuration php-yoshi

Fixes #1007

### CODE REVIEW

**Review by bcoe — CHANGES_REQUESTED** [review]
This work looks really solid to me, one small question, curious why you opted to change the name of the `RootComposer` to `RootComposerUpdatePackages`.

This is out of curiosity, non-blocking.

File: test/releasers/php.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    getFileContentsStub
+      .withArgs('PubSub/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "pubsub"}'));
+    getFileContentsStub
+      .withArgs('PubSub/VERSION', 'master')
+      .resolves(buildGitHubFileRaw('1.0.1'));
+    getFileContentsStub
+      .withArgs('Speech/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "speech"}'));
+    getFileContentsStub
```
Comment (bcoe) [inline test/releasers/php.ts:106]: It seems like we're recreating my team's use-case in this test, which is a PHP library with many composer files in it.

Would it make sense for us to have a more concise unit test, that just updates a root level composer file?

File: test/releasers/php.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      .withArgs('composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"replace": {}}'));
+    getFileContentsStub
+      .withArgs('docs/manifest.json', 'master')
+      .resolves(
+        buildGitHubFileRaw(
+          '{"modules": [{"name": "google/cloud", "versions": []}, {"name": "datastore", "versions": []}]}'
+        )
+      );
+    getFileContentsStub.rejects(
```
Comment (bcoe) [inline test/releasers/php.ts:125]: I'm not sure what this 404 correlates too, perhaps trimming down the unit test to the simple PHP case we can drop this.

File: src/releasers/php.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+// PHP Specific.
+import {RootComposerUpdatePackage} from '../updaters/root-composer-update-package';
+import {ReleasePRConstructorOptions} from '..';
+
+const CHANGELOG_SECTIONS = [
+  {type: 'feat', section: 'Features'},
+  {type: 'fix', section: 'Bug Fixes'},
+  {type: 'perf', section: 'Performance Improvements'},
+  {type: 'revert', section: 'Reverts'},
+  {type: 'docs', section: 'Documentation'},
```
Comment (bcoe) [inline src/releasers/php.ts:29]: Non blocking, do you think that folks in the PHP community will want to create releases when they create doc updates?

For the Node.js releaser we opt to only release on `fix`, `feat`, and breaking changes.

File: src/releasers/php-yoshi.ts [hunk 4]
```diff
@@ -87,7 +87,7 @@ export class PHPYoshi extends ReleasePR {
     // update the aggregate package information in the root
     // composer.json and manifest.json.
     updates.push(
-      new RootComposer({
+      new RootComposerUpdatePackages({
```
Comment (bcoe) [inline src/releasers/php-yoshi.ts:90]: I'm fine with this rename, was just curious the motivation.

**Review by bcoe — CHANGES_REQUESTED** [review]
Reading through your TypeScript, I never would have known this was your first time writing it, you followed the structure of the existing codebase quite well.

I think to land this, the only change I'd like to see would be us trimming down the test case to a really simple PHP repo with a single composer file.

**Comment by setnemo** [comment]
Thanks, I need to update my pull request according to the comments. Project's codebase is pretty structured and I have been writing in OOP style for a long time (php). 

I renamed it because I hope that the new name said "I am updating packages", and my class name said "I am updating a package". I think the correct names is very important, and if you see the code for the first time, correct names helps to understand the code. My English is not very good, maybe that's why the name seemed strange to you :) Help me please with renaming this classes :)

**Comment by bcoe** [comment]
@setnemo 👍 thanks for explanation of variable rename, I'm happy with the new name.

The only blocker on this PR is that I would like to simplify `test/releasers/php.ts` to be a simpler PHP module, so that we can mock a few less files in the test.

Let me know if you need any help, I'm excited to merge this.

File: src/releasers/php.ts (hunk 3 above)
Comment (setnemo) [reply src/releasers/php.ts:29]: Yes, I need to add hidden: true

File: test/releasers/php.ts (hunk 2 above)
Comment (setnemo) [reply test/releasers/php.ts:89]: removed

File: test/releasers/php.ts (hunk 1 above)
Comment (setnemo) [reply test/releasers/php.ts]: You are absolutely right. I use copy/paste and refactor existing tests for my case. I agree with you, so I removed a lot of lines that duplicate test case with yoshi style.

**Comment by setnemo** [comment]
@bcoe I fixed everything, I guess. Check, please.

File: test/releasers/php.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    const req = nock('https://api.github.com')
+      // now we fetch the commits via the graphql API;
+      // note they will be truncated to just before the tag's sha.
+      .post('/graphql', () => {
+        return true;
+      })
+      .reply(200, {
+        data: graphql,
+      });
+    console.log(nock.activeMocks());
```
Comment (bcoe) [inline test/releasers/php.ts:78]:
```suggestion
```

**Comment by setnemo** [comment]
@googlebot I consent.

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@setnemo thank you for the contribution 👏

**Pull request merged**